frontend/src/features/permits/pages/Application/components/dashboard/StartApplicationAction.tsx

```diff
@@ -1,56 +1,161 @@
 
   const handleStartButtonClicked = () => {
     if (chooseFrom !== EMPTY_PERMIT_TYPE_SELECT) {
       navigate(APPLICATIONS_ROUTES.START_APPLICATION(chooseFrom));
+    } else {
+      setIsError(true);
     }
   };
 
+  // Update the structure of menuItems to ensure the callback is applied correctly
+  const menuItems = ALL_PERMIT_TYPE_CHOOSE_FROM_OPTIONS.map(
+    (item: PermitTypeChooseFromItem) => ({
+      ...item,
+      callback: (event: React.MouseEvent<HTMLElement>) =>
+        handleChooseFrom(event, item),
+      // Correctly set the nested item's callback
+      items: item?.items?.map((nestedItem) => ({
+        ...nestedItem,
+        callback: (event: React.MouseEvent<HTMLElement>) =>
+          handleChooseFrom(event, nestedItem),
+      })),
+    }),
+  );
+
+  const [anchorEl, setAnchorEl] = useState<HTMLDivElement | null>();
+  const open = Boolean(anchorEl);
+
+  const handleClick = (e: any) =>
+    setAnchorEl(e.currentTarget as HTMLDivElement);
+  const handleClose = () => setAnchorEl(null);
+
   return (
     <Box className="start-application-action">
-      <SelectPermitType
-        value={chooseFrom}
-        label={"Select Permit Type"}
-        onChange={handleChooseFrom}
-        menuItems={PERMIT_TYPE_CHOOSE_FROM_OPTIONS.map((data) => (
-          <MenuItem key={data.value} value={data.value}>
-            {data.label}
-          </MenuItem>
-        ))}
-      />
-
-      <Button
-        className="start-application-action__btn"
-        variant="contained"
-        onClick={handleStartButtonClicked}
-      >
-        Start Application
-      </Button>
+      <FormLabel className="start-application-action__label">
+        Select Permit Type
+      </FormLabel>
+      <div className="start-application-action__control">
+        <Tooltip
+          className="start-application-action__input-tooltip"
+          title={
+            chooseFrom !== EMPTY_PERMIT_TYPE_SELECT
+              ? getFormattedPermitTypeName(chooseFrom)
+              : EMPTY_PERMIT_TYPE_SELECT
+          }
+        >
+          <Button
+            className={`start-application-action__input ${open && "start-application-action__input--open"} ${isError && "start-application-action__input--error"}`}
+            onClick={handleClick}
+          >
+            <span className="start-application-action__input-text">
+              {chooseFrom !== EMPTY_PERMIT_TYPE_SELECT
+                ? getFormattedPermitTypeName(chooseFrom)
+                : EMPTY_PERMIT_TYPE_SELECT}
+            </span>
+            <FontAwesomeIcon
+              className="start-application__input-icon"
+              icon={faChevronDown}
+            />
+          </Button>
+        </Tooltip>
+
+        <Menu
+          anchorEl={anchorEl}
+          onClose={handleClose}
+          open={open}
+          slotProps={{
+            paper: {
+              className: `start-application-action__menu-container ${open && "start-application-action__menu-container--open"}`,
```

Comment:
Please use proper ternary syntax here, and also consider extracting this long className into a variable